tools/StaticAnalysis/BreakingChangeAnalyzer/BreakingChangeAnalyzer.cs

```diff
@@ -0,0 +1,968 @@
+        {
+            // If the old cmdlet implements SupportsShouldProcess and the new cmdlet does not, log an issue
+            if (oldCmdlet.SupportsShouldProcess && !newCmdlet.SupportsShouldProcess)
+            {
+                issueLogger.LogBreakingChangeIssue(
+                    cmdlet: oldCmdlet,
+                    severity: 0,
+                    problemId: ProblemIds.BreakingChangeProblemId.RemovedShouldProcess,
+                    description: string.Format(Properties.Resources.RemovedShouldProcessDescription, oldCmdlet.Name),
+                    remediation: string.Format(Properties.Resources.RemovedShouldProcessRemediation, oldCmdlet.Name));
+            }
+        }
+
+        /// <summary>
+        /// Check if a cmdlet no longer implements SupportsPaging.
+        /// </summary>
+        /// <param name="oldCmdlet"></param>
+        /// <param name="newCmdlet"></param>
+        /// <param name="issueLogger"></param>
+        private void CheckForRemovedSupportsPaging(
+            CmdletBreakingChangeMetadata oldCmdlet,
+            CmdletBreakingChangeMetadata newCmdlet,
+            ReportLogger<BreakingChangeIssue> issueLogger)
+        {
+            // If the old cmdlet implements SupportsPaging and the new cmdlet does not, log an issue
+            if (oldCmdlet.SupportsPaging && !newCmdlet.SupportsPaging)
+            {
+                issueLogger.LogBreakingChangeIssue(
+                    cmdlet: oldCmdlet,
+                    severity: 0,
+                    problemId: ProblemIds.BreakingChangeProblemId.RemovedPaging,
+                    description: string.Format(Properties.Resources.RemovedPagingDescription, oldCmdlet.Name),
+                    remediation: string.Format(Properties.Resources.RemovedPagingRemediation, oldCmdlet.Name));
+            }
+        }
+
+        /// <summary>
+        /// Check if the OutputType of the cmdlet has been removed, or if any property
+        /// of the OutputType has been removed or changed
+        /// </summary>
+        /// <param name="oldCmdlet"></param>
+        /// <param name="newCmdlet"></param>
+        /// <param name="issueLogger"></param>
+        private void CheckForChangedOutputType(
+            CmdletBreakingChangeMetadata oldCmdlet,
+            CmdletBreakingChangeMetadata newCmdlet,
+            ReportLogger<BreakingChangeIssue> issueLogger)
+        {
+            // For each output in the old cmdlet assembly, look for the corresponding
+            // output in the new cmdlet assembly
+            foreach (var oldOutput in oldCmdlet.OutputTypes)
+            {
+                bool found = false;
+
+                foreach (var newOutput in newCmdlet.OutputTypes)
+                {
+                    // If the corresponding output has been found in the new assembly,
+                    // look for through all of the properties
+                    if (oldOutput.Type.Name.Equals(newOutput.Type.Name))
```

Comment:
Hmm, so I think this would be a little smoother and easier to follow  if:

(1) You had a dictionary of output types, keyed by output type name.  You could then just look for the key in the dictionary 
(2) If there is no match, use Linq to cimplify your code - if you have a simple predocate that does recursive property patching, you can just call Any<t>(fiulter) on the list of values to find if any outpout types match
(3) If you put the matching function into the output types themselves, it will separate concers a little better and clean up the look of this code.